dpsim/include/dpsim/KLUAdapter.h

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2017-2021 Institute for Automation of Complex Power Systems,
+ *                     EONERC, RWTH Aachen University
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *********************************************************************************/
+
+#pragma once
+
+extern "C"
+{
+    #include <klu.h>
+}
+
+#include <iostream>
+#include <vector>
+#include <list>
+#include <unordered_map>
+#include <bitset>
+#include <memory>
+#include <fstream>
+
+#include <dpsim/Config.h>
+#include <dpsim/Definitions.h>
+#include <dpsim/DirectLinearSolver.h>
+
+namespace DPsim
+{
+
+    class KLUAdapter : public DirectLinearSolver
+    {
+        /// Vector of variable entries in system matrix
+        std::vector<std::pair<UInt, UInt>> changedEntries;
+
+		/// KLU-specific structs
+        klu_common m_common;
+		klu_numeric* m_numeric = nullptr;
+		klu_symbolic* m_symbolic = nullptr;
+
+		/// Flags to indicate mode of operation
+		/// Define which ordering to choose in preprocessing
+		int m_ordering = KLU_AMD_FP;
+
+		/// Use BTF or not
+		int m_btf = 1;
+
+		/// Define scaling method
+		/// Options are: 1: sum-scaling, 2: max scaling, <=0: no scaling
+		int m_scaling = 1;
```

Comment:
`enum` used, not `enum class`. Otherwise, the entries `NO_SCALING...` cannot be casted to an integer `m_scaling`. This can be changed to an `enum class` (maybe a member of config-class?) later and handled by each individual Adapter